src/pages/teachers/curriculum/sitemap.xml/index.ts

```diff
@@ -0,0 +1,65 @@
+import path from "node:path/posix";
+
+import { GetServerSideProps } from "next";
+import { getServerSideSitemap } from "next-sitemap";
+
+import curriculumApi2023 from "@/node-lib/curriculum-api-2023";
+
+/**
+ * Get all curriculum pages and construct sitemap entries for them.
+ */
+
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  const sitemapBaseUrl = process.env.SITEMAP_BASE_URL || "";
+
+  const basePath = "teachers/curriculum";
+
+  const curriculumResult = await curriculumApi2023.subjectPhaseOptions();
+
+  const curriculumPathSlugs: string[] = [];
+
+  for (const curriculum of curriculumResult) {
+    curriculumPathSlugs.push(`${curriculum.slug}-primary`);
+    if (curriculum.examboards) {
```

Comment:
This will miss secondary subjects without examboards, so I think we need an `else` case for when there aren't examboards but there's still a secondary phase.